test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/jbpm/ProcessFailoverIntegrationTest.java

```diff
@@ -43,92 +55,207 @@
+        Instance kieServerInstance = deploymentScenario.getKieServerDeployments().get(0).getInstances().iterator().next();
 
-        logger.debug("Start process instance");
-        Long longScriptPid = processServicesClient.startProcess(CONTAINER_ID, Constants.ProcessId.LONG_SCRIPT, Collections.emptyMap());
-        assertThat(longScriptPid).isNotNull().isGreaterThan(0L);
-        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+        Long longScriptPid = startProcessInstance(Constants.ProcessId.LONG_SCRIPT);
 
         assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
-        assertProcessVariable(longScriptPid, variableKey, variableValueOne);
+        assertProcessVariable(longScriptPid, VARIABLE_KEY, VARIABLE_VALUE_ONE);
 
         logger.debug("Send signal to continue with process.");
         signalStartLongScript(longScriptPid);
 
         logger.debug("Force delete (Kill) Kie server instance.");
-        deploymentScenario.getKieServerDeployment().deleteInstances(kieServerInstance);
+        deploymentScenario.getKieServerDeployments().get(0).deleteInstances(kieServerInstance);
         logger.debug("Wait for scale");
-        deploymentScenario.getKieServerDeployment().waitForScale();
+        deploymentScenario.getKieServerDeployments().get(0).waitForScale();
 
         logger.debug("Send signal to try complete process. Not able yet.");
         processServicesClient.signalProcessInstance(CONTAINER_ID, longScriptPid, Constants.Signal.SIGNAL_2_NAME, null);
 
         assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
-        assertProcessVariable(longScriptPid, variableKey, variableValueOne);
+        assertProcessVariable(longScriptPid, VARIABLE_KEY, VARIABLE_VALUE_ONE);
 
         logger.debug("Send signal again to continue with process. It was rollbacked.");
         processServicesClient.signalProcessInstance(CONTAINER_ID, longScriptPid, Constants.Signal.SIGNAL_NAME, null);
 
         assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
-        assertProcessVariable(longScriptPid, variableKey, variableValueTwo);
+        assertProcessVariable(longScriptPid, VARIABLE_KEY, VARIABLE_VALUE_TWO);
 
         processServicesClient.signalProcessInstance(CONTAINER_ID, longScriptPid, Constants.Signal.SIGNAL_2_NAME, null);
         assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED);
 
-        assertThat(deploymentScenario.getKieServerDeployment().getInstances().iterator().next()).isNotEqualTo(kieServerInstance);
+        assertThat(deploymentScenario.getKieServerDeployments().get(0).getInstances().iterator().next()).isNotEqualTo(kieServerInstance);
+    }
+
+    @Test
+    public void processFailoverRedirectionTest() {
```

Comment:
What is the purpose of this test?

Reply:
In this test there are 2 Kie Server and Kie Server with running process (used script with thread sleep) is killed. Process from the killed Kie Server is completed on the second Kie Server instance (pod).